Copied from upstream:

https://github.com/LubosD/twinkle/commit/4b42755619011c117a76bdf98e417ebedc47e319

From 4b42755619011c117a76bdf98e417ebedc47e319 Mon Sep 17 00:00:00 2001
From: Michal Kubecek <mkubecek@suse.cz>
Date: Wed, 6 Jun 2018 10:07:21 +0200
Subject: [PATCH] Include <QRegExpValidator> explicitly

Since Qt 5.11, generated ui_getprofilename.h no longer includes QHeaderView
which breaks the chain that included qvalidator.h in getprofilename.cpp.
As it feels rather fragile to rely on such indirect includes, let's include
<QRegExpValidator> explicitly in each file using QRegExpValidator class.
---
 src/gui/diamondcardprofileform.cpp | 1 +
 src/gui/getprofilenameform.cpp     | 2 +-
 src/gui/inviteform.cpp             | 1 +
 src/gui/mphoneform.cpp             | 1 +
 src/gui/numberconversionform.cpp   | 1 +
 src/gui/syssettingsform.cpp        | 1 +
 src/gui/userprofileform.cpp        | 1 +
 src/gui/wizardform.cpp             | 1 +
 8 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/gui/diamondcardprofileform.cpp b/src/gui/diamondcardprofileform.cpp
index 6656909..517180b 100644
--- a/src/gui/diamondcardprofileform.cpp
+++ b/src/gui/diamondcardprofileform.cpp
@@ -21,6 +21,7 @@
 
 #include <QRegExp>
 #include <QValidator>
+#include <QRegExpValidator>
 #include "gui.h"
 #include "diamondcard.h"
 #include "getprofilenameform.h"
diff --git a/src/gui/getprofilenameform.cpp b/src/gui/getprofilenameform.cpp
index 1319e1d..89c715e 100644
--- a/src/gui/getprofilenameform.cpp
+++ b/src/gui/getprofilenameform.cpp
@@ -1,7 +1,7 @@
 #include "getprofilenameform.h"
-
 #include <QDir>
 #include <QMessageBox>
+#include <QRegExpValidator>
 #include "user.h"
 #include "protocol.h"
 
diff --git a/src/gui/inviteform.cpp b/src/gui/inviteform.cpp
index 433fb22..2a5b68d 100644
--- a/src/gui/inviteform.cpp
+++ b/src/gui/inviteform.cpp
@@ -7,6 +7,7 @@
 #include "sys_settings.h"
 #include <QRegExp>
 #include <QValidator>
+#include <QRegExpValidator>
 
 /*
     Copyright (C) 2005-2009  Michel de Boer <michel@twinklephone.com>
diff --git a/src/gui/mphoneform.cpp b/src/gui/mphoneform.cpp
index 260fda7..c4e3c1d 100644
--- a/src/gui/mphoneform.cpp
+++ b/src/gui/mphoneform.cpp
@@ -54,6 +54,7 @@
 #include <QRegExp>
 #include <QValidator>
 #include <QSettings>
+#include <QRegExpValidator>
 #include "buddyform.h"
 #include "diamondcardprofileform.h"
 #include "osd.h"
diff --git a/src/gui/numberconversionform.cpp b/src/gui/numberconversionform.cpp
index f8ae64c..8481a9b 100644
--- a/src/gui/numberconversionform.cpp
+++ b/src/gui/numberconversionform.cpp
@@ -1,5 +1,6 @@
 #include "numberconversionform.h"
 
+#include <QRegExpValidator>
 #include "gui.h"
 
 /*
diff --git a/src/gui/syssettingsform.cpp b/src/gui/syssettingsform.cpp
index 216af54..355df59 100644
--- a/src/gui/syssettingsform.cpp
+++ b/src/gui/syssettingsform.cpp
@@ -28,6 +28,7 @@
 #include "twinkle_config.h"
 #include <QRegExp>
 #include <QValidator>
+#include <QRegExpValidator>
 #include "syssettingsform.h"
 /*
  *  Constructs a SysSettingsForm as a child of 'parent', with the
diff --git a/src/gui/userprofileform.cpp b/src/gui/userprofileform.cpp
index 28700a6..9ed9209 100644
--- a/src/gui/userprofileform.cpp
+++ b/src/gui/userprofileform.cpp
@@ -31,6 +31,7 @@
 #include <QStringList>
 #include "twinkle_config.h"
 #include <QListWidget>
+#include <QRegExpValidator>
 #include "numberconversionform.h"
 #include "util.h"
 #include "userprofileform.h"
diff --git a/src/gui/wizardform.cpp b/src/gui/wizardform.cpp
index 777aa12..f925875 100644
--- a/src/gui/wizardform.cpp
+++ b/src/gui/wizardform.cpp
@@ -23,6 +23,7 @@
 #include <QTextStream>
 #include "gui.h"
 #include <QFile>
+#include <QRegExpValidator>
 #include "wizardform.h"
 
 #define PROV_NONE	QT_TRANSLATE_NOOP("WizardForm", "None (direct IP to IP calls)")
-- 
2.17.0

